### PR TITLE
feat: add `prop`, `props` and `always` functions

### PR DESCRIPTION
**Pull request by gabriel-ss**
Addresses #200. I tried to make the implementations close to the ones present in [ramda](https://ramdajs.com) and the type inference as precise as possible, but I could tweak it. Let me know if I could make any changes to make the PR more fitting for the lib.

### CODE REVIEW

**Comment by ppeeou** [comment]
thank you for your contribution. we will review it :)

**Comment by ppeeou** [comment]
@gabriel-ss 
if you add only type tests ([type-check folder](https://github.com/marpple/FxTS/tree/main/type-check)), it will be okay to merge pr👍

Could you add type tests?

**Comment by gabriel-ss** [comment]
Sure! Didn't notice the folder when I was preparing the PR; I'll look into it over the weekend and let you know as soon as it's done.

**Comment by gabriel-ss** [comment]
@ppeeou Done!

**Review by shine1594 — APPROVED** [review]
Thanks for your contribution 👍

File: type-check/props.test.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  ],
+  map(props(["str", "num"] as const)),
+  toArray,
+);
+
+checks([
+  check<typeof res1, ["value"], Test.Pass>(),
+  check<typeof res2, ["value"], Test.Pass>(),
+  check<typeof res3, undefined[], Test.Pass>(),
+  check<typeof res4, undefined[], Test.Pass>(),
```
[the remark this answers is not in the document]
Comment (gabriel-ss) [reply type-check/props.test.ts]: Well noted, made a new commit updating it.

**Pull request merged**